web/profiles/custom/yalesites_profile/modules/custom/ys_core/src/SearchBreadcrumbTwigExtension.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Drupal\ys_core;
+
+use Twig\Extension\AbstractExtension;
+use Twig\TwigFunction;
+use Drupal\Core\Routing\RouteMatch;
+
+/**
+ * Twig functions to retrieve breadcrumbs for a specific node.
+ *
+ * @todo This extension is not complete and more work needs to be done here.
+ * Main issue is that the core breadcrumb service seems to ignore the passed
+ * route parameter so getting breadcrumbs for a specific node does not work.
+ * Instead, it only works with the current route. More details and a possible
+ * fix here: https://drupal.stackexchange.com/questions/191548/how-do-i-build-breadcrumbs-for-a-certain-node
+ * and here: https://www.drupal.org/project/menu_breadcrumb/issues/3026188
+ */
+class SearchBreadcrumbTwigExtension extends AbstractExtension {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFunctions() {
+    return [
+      new TwigFunction('getBreadcrumbsFromNid', [$this, 'getBreadcrumbsFromNid']),
+    ];
+  }
+
+  /**
+   * Function that returns breadcrumbs for a specified node ID.
+   *
+   * @param string $nid
+   *   Node ID.
+   */
+  public function getBreadcrumbsFromNid($nid) {
+    $links = [];
+
+    if (!empty($nid)) {
+      // Is this node in the menu?
+      $menu_link_manager = \Drupal::service('plugin.manager.menu.link');
+      $result = $menu_link_manager->loadLinksByRoute('entity.node.canonical', ['node' => $nid]);
+
+      if (!empty($result)) {
+        // Get the breadcrumbs for this node.
+        $entity = \Drupal::entityTypeManager()->getStorage('node')->load($nid);
+        $routeName = $entity->toUrl()->getRouteName();
+        $routeParameters = $entity->toUrl()->getRouteParameters();
+        $route = \Drupal::service('router.route_provider')->getRouteByName($routeName);
+        $routeMatch = new RouteMatch($routeName, $route, $routeParameters, $routeParameters);
+
+        $breadcrumbs = \Drupal::service('breadcrumb')->build($routeMatch)->getLinks();
+
+        // // Process all breadcrumbs.
+        // foreach ($breadcrumbs['#links'] as $breadcrumb) {
+        //   array_push($links, [
+        //     'title' => $breadcrumb->getText(),
+        //     'url' => $breadcrumb->getUrl()->toString(),
+        //     'is_active' => empty($breadcrumb->getUrl()->toString()),
+        //   ]);
+        // }
+        // return $links;
```

Comment:
If we're not using this, it should be deleted.

Reply:
We can remove, but this file is currently a work in progress to get the breadcrumbs for search results. Jim said if there wasn't anything breaking, to leave the work in place and I've added a @todo to it and a ticket in Jira to return to this work at some point.

Reply:
It was causing more issues in linting, so I removed this altogether.